btcjson/chainsvrresults.go

```diff
@@ -379,6 +382,48 @@ type GetRawMempoolVerboseResult struct {
 	Depends          []string `json:"depends"`
 }
 
+// ZmqNotificationResult models the data returned from the zmqnotifications command.
+type ZmqNotificationResult struct {
+	Type          string
+	Address       *url.URL
+	HighWaterMark int
+}
+
+func (z *ZmqNotificationResult) MarshalJSON() ([]byte, error) {
+	return json.Marshal(map[string]interface{}{
+		"type":    z.Type,
+		"address": z.Address.String(),
+		"hwm":     z.HighWaterMark,
+	})
+}
+
+// UnmarshalJSON satisfies the json.Unmarshaller interface
+func (z *ZmqNotificationResult) UnmarshalJSON(bytes []byte) error {
+	type basicNotification struct {
+		Type    string
+		Address string
+		Hwm     int
+	}
```

Comment:
We can use this in the `MarshalJSON` method as well, as a replacement for `map[string]interface{}` type, by moving `basicNotification` outside.

What do you think?

Reply:
A benefit of keeping `basicNotification` within the unmarshaling method is that it doesn't pollute the package namespace. So I think it's better to keep it where it is now